Week 2 Section 2

DvdHalliday opened the pull request:
I started making this section before we received the coding standards material so it wasn't made according to them.
Here are some pics of the project behaviour: 



![e1](https://user-images.githubusercontent.com/109089386/179550053-39167954-0c32-47b5-9514-4085e72bff0c.png)
![e2](https://user-images.githubusercontent.com/109089386/179550056-ea9323cb-cc18-4234-85b8-7e9a2aa4685b.png)
![e3](https://user-images.githubusercontent.com/109089386/179550057-ecdf7483-27c8-478e-a72a-9766941fbc1e.png)
![e4](https://user-images.githubusercontent.com/109089386/179550059-fbc38326-5484-4b92-a8ce-5caf858e722d.png)
![e5](https://user-images.githubusercontent.com/109089386/179550062-806f2d0e-2a8c-4d70-a9d7-157b198bc3da.png)
![e6](https://user-images.githubusercontent.com/109089386/179550063-c24251c9-e9ea-41dd-af10-f461b5db2469.png)

DvdHalliday committed:
Exercise 1

Uploading both files

DvdHalliday committed:
Exercise 2

Uploading the Playlist class files

DvdHalliday committed:
Exercise 3

PlaylistContainer class files

DvdHalliday committed:
Exercise 4.0

Uploading both files according to requirements, the Spotify.cpp's still a work in progress

DvdHalliday committed:
Exercise 4.1

[WORK IN PROGRESS] Uploading both files, sort of tested it's functionality to check if they work as they should and it's looking good so far

DvdHalliday committed:
Exercise 4.2 & small fix to Playlist.cpp

Added both spotify header and cpp files according to the exercise needs.
I also modified a small detail in the Playlist cpp because I noticed it was printing just the duration and not the formatted one.

DvdHalliday committed:
Exercise 4.3 + final updates

Here are both spotify files up to date according to this exercise.
I am also uploading the .sln and main.cpp files

DvdHalliday committed:
Fixing input management in the CreatePlaylist menu

There was some trouble with the way I was receiving the inputs, I was not taking the whole line, instead I was just picking the first string before a space, so, in order to get that whole name the user is entering, I had to change some stuff in that function.

jeroo97 commented on c++ training/W2S2/Spotify/Song.cpp:
```diff
@@ -0,0 +1,66 @@
+	m_duration = 125;
+}
+
+Song::Song(std::string songTitle, std::string artistName, int duration)
```
I think these parameters could be `const`

jeroo97 replied:
And the same comment for the rest of the individual setters.

jeroo97 commented on c++ training/W2S2/Spotify/Song.cpp:
```diff
@@ -0,0 +1,66 @@
+std::string Song::getFormattedDuration() const
+{
+	std::string formattedString = "";
+	int seconds = m_duration % 60;
```
These two variables (seconds & minutes) could also be `const` because you never modify them.

jeroo97 commented on c++ training/W2S2/Spotify/Playlist.cpp:
```diff
@@ -0,0 +1,45 @@
+	m_songCount = 0;
+}
+
+Playlist::Playlist(std::string playlistTitle)
```
The parameter could be `const`, as well as the individual setters.

jeroo97 commented on c++ training/W2S2/Spotify/PlaylistContainer.cpp:
```diff
@@ -0,0 +1,22 @@
+	m_playlistCount++;
+}
+
+Playlist PlaylistContainer::getPlaylist(int playlistIndex) const
```
This could be a `const` parameter

JDAlzate commented on c++ training/W2S2/Spotify/Song.cpp:
```diff
@@ -0,0 +1,66 @@
+
+bool Song::isValid() const
+{
+	return (m_duration>0 && !empty(m_artistName) && !empty(m_songTitle));
```
Think about these validation parameters when thinking about your default values. The default constructor initializes your members to arbitrary, valid parameters. That means that it would be acceptable to enter a default-constructed song into a playlist, is that what you want?

JDAlzate commented on c++ training/W2S2/Spotify/main.cpp:
```diff
@@ -0,0 +1,7 @@
+#include "Spotify.h";
+
+int main() {
+	Spotify spot = Spotify();
```
There is no need to assign any value to spot. Writing `Spotify spot;` will call the default constructor. Also, spot is an ambiguous name, do you really need to abbreviate spotify?

JDAlzate commented on c++ training/W2S2/Spotify/Spotify.cpp:
```diff
@@ -0,0 +1,177 @@
+	std::cout << "Please enter a name for the playlist: ";
+	std::string playlistName;
+	std::getline(std::cin,playlistName);
+	//std::cin >> playlistName;
```
Why do you need to keep this comment here?

JDAlzate commented on c++ training/W2S2/Spotify/Spotify.cpp:
```diff
@@ -0,0 +1,177 @@
+
+void Spotify::openMainMenu()
+{
+	system("cls");
```
Spaces are your friends. Good use of spaces can make the code more legible by separating the code into similar actions/ideas. Having this huge method with no spaces between statements makes it tedious to read.

JDAlzate commented on c++ training/W2S2/Spotify/Spotify.cpp:
```diff
@@ -0,0 +1,177 @@
+
+void Spotify::runApp()
+{
+	while (true) {//m_activeCommand != MenuCommand::Exit
```
Why do you have this in a comment? Why not make it the actual condition?

JDAlzate reviewed (changes requested):
Review is up!

DvdHalliday committed:
Reviews update

Added const to all the parameters and local variables that needed it, changed spot's name to spotify, added some spacing to the spotify cpp methods and removed some forgotten comments.

JDAlzate reviewed (approved):
Merge approved!

JDAlzate reviewed (approved):
Approve test
